source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -287,16 +287,18 @@ uint16_t InstanceImpl::ThreadLocalPool::shardSize() {
   }
 
   Common::Redis::RespValue request;
-  for (uint16_t size = 0;; size++) {
+  std::set<Upstream::HostConstSharedPtr> uniqueHosts;
```

Comment:
std::unordered_set would be more appropriate for this. Also, our coding style prefers snake case for variable names.

Since we know the upper bound on the number of hosts, we'll want something like:
```suggestion
  std::unordered_set<Upstream::HostConstSharedPtr> unique_hosts;
  unique_hosts.reserve(Envoy::Extensions::Clusters::Redis::MaxSlot);
```

Reply:
Thank you for pointing this out! I have replaced std::set with absl::flat_hash_set as it is preferred over std::unordered_set for performance reasons. I have also added .reserve() to pre-allocate memory based on the known upper bound as you suggested.